src/nvim/api/vimscript.c

```diff
@@ -39,13 +39,25 @@
 /// @see |nvim_cmd()|
 ///
 /// @param src      Vimscript code
-/// @param output   Capture and return all (non-error, non-shell |:!|) output
+/// @param opts  Optional parameters.
+///         - output: Capture and return all (non-error, non-shell |:!|) output.
+///           Default: `false`.
 /// @param[out] err Error details (Vim error), if any
-/// @return Output (non-error, non-shell |:!|) if `output` is true,
+/// @return Output (non-error, non-shell |:!|) if `opts.output` is true,
 ///         else empty string.
-String nvim_exec(uint64_t channel_id, String src, Boolean output, Error *err)
-  FUNC_API_SINCE(7)
+String nvim_exec2(uint64_t channel_id, String src, Dict(exec) *opts, Error *err)
+  FUNC_API_SINCE(10)
 {
+  Boolean output = false;
+  if (HAS_KEY(opts->output)) {
+    if (opts->output.type != kObjectTypeBoolean) {
+      api_set_error(err, kErrorTypeValidation, "'output' should be boolean");
+      return (String)STRING_INIT;
+    }
+
+    output = opts->output.data.boolean;
+  }
+
   const int save_msg_silent = msg_silent;
```

Comment:
using `TRY_WRAP` may improve exception behavior. Don't need to make this optional.

see other functions in this file for `TRY_WRAP` example usage.

Reply:
@justinmk, didn't manage to find a way to do this. More thoughts in a [recent comment](https://github.com/neovim/neovim/pull/19032#issuecomment-1206200586).